### PR TITLE
Make sure `SseEmitter` with heartbeat terminates

### PR DESCRIPTION
**Pull request by Stephan202**

### CODE REVIEW

File: rxjava-support/src/main/java/tech/picnic/rx/RxSpring4Util.java [hunk 1]
```diff
@@ -320,10 +320,14 @@ private RxSpring4Util() {}
       Duration heartbeat,
       LongFunction<T> heartbeatMapping,
       Scheduler scheduler) {
-    return flowable ->
-        Flowable.interval(heartbeat.toMillis(), MILLISECONDS, scheduler)
-            .map(heartbeatMapping::apply)
-            .mergeWith(flowable)
+    return publisher ->
+        Flowable.<T>fromPublisher(publisher)
```
Comment (philleonard) [inline rxjava-support/src/main/java/tech/picnic/rx/RxSpring4Util.java]: Nice, this issue was in the downcasting of `Flowable` -> `Publisher` in the function signature?

Comment (Stephan202) [reply rxjava-support/src/main/java/tech/picnic/rx/RxSpring4Util.java]: Nope, the issue was that `mergeWith` doesn't complete until both sources complete. The `fromPublisher` here allows the source to be viewed as a `Flowable` in the two places that require this.

(In my testing the `#publish` call didn't seem necessary, but this way it is clearer that both the `mergeWith` and `takeUntil` operate on the same set of emissions.)

Comment (philleonard) [reply rxjava-support/src/main/java/tech/picnic/rx/RxSpring4Util.java]: Just thinking, won't this incur multiple subscriptions on `flowable` (the second in `takeUntil`) given that it's emissions aren't replayed?

Comment (philleonard) [reply rxjava-support/src/main/java/tech/picnic/rx/RxSpring4Util.java]: My bad, `publish` covers this :)

Comment (philleonard) [reply rxjava-support/src/main/java/tech/picnic/rx/RxSpring4Util.java]: My bad, this is covered by the `publish`

**Pull request merged**